### PR TITLE
Fixes #2720: Documented the files in openfood/jobs package

### LINKED ISSUE #2720: Document the files in openfood/jobs package
**Summary**

Document the files in openfood/jobs package: [https://github.com/openfoodfacts/openfoodfacts-androidapp/tree/master/app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs](https://github.com/openfoodfacts/openfoodfacts-androidapp/tree/master/app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs)

### PR DESCRIPTION
**Pull request by RajaVamsi11**
## Description

Documented the files in openfood/jobs package

## Related issues and discussion
Fixes #2720

### CODE REVIEW

**Comment by RajaVamsi11** [comment]
@deniger @teolemon Please review this PR.

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/FileDownloader.java [hunk 1]
```diff
@@ -13,23 +13,48 @@
...
+     * A method to download the file using fileUrl and to callback
+     * the FileReceiver interface method fileDownloaded.
+     * @param fileUrl provides the URL of the file to download.
+     * @param callback is used to call the Interface method.
+     */
     public void download(String fileUrl, FileReceiver callback) {
         OpenFoodAPIService client = CommonApiManager.getInstance().getOpenFoodApiService();
         final Call<ResponseBody> responseBodyCall = client.downloadFile(fileUrl);
         responseBodyCall.enqueue(new Callback<ResponseBody>() {
+            /**
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/FileDownloader.java]: Is this really needed? 😅

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/FileDownloader.java [hunk 2]
```diff
@@ -13,23 +13,48 @@
...
+             * An overridden method in which response of the call will be found.
+             *
+             * @param call : call to API to download file
+             * @param response : The result of the Task
+             */
             @Override
             public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+
+                /**
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/FileDownloader.java]: Just insert a normal block comment

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/FileDownloader.java [hunk 3]
```diff
@@ -42,13 +67,24 @@ public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response)
                 }
             }
 
+            /**
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/FileDownloader.java]: Same as before, really needed?

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/FileDownloader.java (hunk 1 above)
Comment (RajaVamsi11) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/FileDownloader.java]: Removed

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/FileDownloader.java (hunk 2 above)
Comment (RajaVamsi11) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/FileDownloader.java]: Used normal block comment

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/FileDownloader.java (hunk 3 above)
Comment (RajaVamsi11) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/FileDownloader.java]: Removed

**Comment by RajaVamsi11** [comment]
@VaiTon I have made the suggested changes. Can you please review again?

**Comment by RajaVamsi11** [comment]
@VaiTon Thanks for approving the PR.

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/FileDownloader.java [hunk 4]
```diff
@@ -13,23 +13,39 @@
...
     private Context context;
 
     public interface FileReceiver {
         void fileDownloaded(File file);
     }
 
+    /**
+     * Constructor of the class used to initialize the objects.
+     * @param context : context to be set
+     */
```
Comment (deniger) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/FileDownloader.java]: please add info about context: could it be null ?
If not add @NonNull annotation.

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/FileDownloader.java [hunk 5]
```diff
@@ -13,23 +13,39 @@
...
     public FileDownloader(Context context) {
         this.context = context;
     }
 
+    /**
+     * A method to download the file using fileUrl and to callback
+     * the FileReceiver interface method fileDownloaded.
+     * @param fileUrl provides the URL of the file to download.
+     * @param callback is used to call the Interface method.
+     */
```
Comment (deniger) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/FileDownloader.java]: same points about nullability of the parameters

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/FileDownloader.java [hunk 6]
```diff
@@ -13,23 +13,39 @@
...
     public FileDownloader(Context context) {
         this.context = context;
     }
 
+    /**
+     * A method to download the file using fileUrl and to callback
+     * the FileReceiver interface method fileDownloaded.
+     * @param fileUrl provides the URL of the file to download.
+     * @param callback is used to call the Interface method.
```
Comment (deniger) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/FileDownloader.java]: callback is called if the file is downloaded with success

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/SavedProductUploadJob.java [hunk 7]
```diff
@@ -6,24 +6,35 @@
 import openfoodfacts.github.scrachx.openfood.network.OpenFoodAPIClient;
 
 /**
- * Created by jayanth on 22/2/18.
+ * Class which is used to upload the saved product by extending Job service
+ * @author jayanth
+ * @since 22/2/18
  */
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/SavedProductUploadJob.java]: Please remove white space before method

Comment (RajaVamsi11) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/SavedProductUploadJob.java]: Done.

**Comment by RajaVamsi11** [comment]
@teolemon @Karljoones @deniger I am not able to solve the conflicts because of the `Resolve Conflicts` button is being disabled. Is there any way to solve the conflicts.

**Comment by Karljoones** [comment]
@RajaVamsi11 the other way to do this would be to rebase your pull request, to ensure that the code is up to date

**Comment by teolemon** [comment]
@vanshikaarora can you rebase ? If no update, I propose closing

**Comment by VaiTon** [comment]
Merged with c5ce54f83bb84632ed4f55131ad38472f41971b6.

**Pull request closed without merging**